pkg/minikube/openshiftversions/openshift_versions.go

```diff
@@ -17,48 +17,125 @@ limitations under the License.
 package openshiftversions
 
 import (
+	"encoding/json"
 	"fmt"
 	"io"
-
-	"github.com/google/go-github/github"
-	"github.com/minishift/minishift/pkg/util"
-	githubutil "github.com/minishift/minishift/pkg/util/github"
-	"github.com/pkg/errors"
+	"net/http"
+	"strings"
+	"time"
 )
 
-const githubOwner = "openshift"
-const githubRepo = "origin"
+type ImageTags struct {
+	Count    int         `json:"count"`
+	Next     string      `json:"next"`
+	Previous interface{} `json:"previous"`
+	Results  []Results   `json:"results"`
+}
+
+type Results struct {
+	Name        string      `json:"name"`
+	FullSize    int         `json:"full_size"`
+	Images      []ImageInfo `json:"images"`
+	ID          int         `json:"id"`
+	Repository  int         `json:"repository"`
+	Creator     int         `json:"creator"`
+	LastUpdater int         `json:"last_updater"`
+	LastUpdated time.Time   `json:"last_updated"`
+	ImageID     interface{} `json:"image_id"`
+	V2          bool        `json:"v2"`
+}
 
-func PrintOpenShiftVersionsFromGitHub(output io.Writer) {
-	PrintOpenShiftVersions(output)
+type ImageInfo struct {
+	Size         int         `json:"size"`
+	Architecture string      `json:"architecture"`
+	Variant      interface{} `json:"variant"`
+	Features     interface{} `json:"features"`
+	Os           interface{} `json:"os"`
+	OsVersion    interface{} `json:"os_version"`
+	OsFeatures   interface{} `json:"os_features"`
 }
 
-func PrintOpenShiftVersions(output io.Writer) {
-	versions, err := getVersions()
+func PrintDownStreamVersions(output io.Writer, minSupportedVersion string) {
```

Comment:
So where is the difference in  _PrintDownStreamVersions_ vs _PrintUpStreamVersions_ apart from the URL they access? Is this not really the same method, just different URL in which case we can just make the URL a method parameter? Probably a good idea to externalise the URL and put them for example into the constant file